staging/src/k8s.io/apiserver/pkg/util/flowcontrol/filter-config.go

```diff
@@ -0,0 +1,361 @@
+			plState.queues = reqMgr.queueSetFactory.NewQueueSet(plState.qsConfig)
+		} else {
+			klog.V(5).Infof("Retaining queues for priority level %q: config=%#+v, concurrencyLimit=%d, quiescent=%v (shares=%v, shareSum=%v)", plName, plState.config, plState.qsConfig.ConcurrencyLimit, plState.emptyHandler != nil, plState.config.Limited.AssuredConcurrencyShares, shareSum)
+			plState.queues.SetConfiguration(plState.qsConfig) // TODO: make sure that retains other config if zero desired queues
+		}
+	}
+	reqMgr.curState.Store(newRMState)
+	klog.V(5).Infof("Switched to new RequestManagementState")
+	// We do the following only after updating curState to guarantee
+	// that if Wait returns `tryAnother==true` then a fresh load from
+	// curState will yield an requestManagerState that is at least
+	// as up-to-date as the data here.
+	//
+	// Put another way: doing the Quiesce calls after Store(X) ensures
+	// that if any call to qs.Quiesce causes a call to qs.Wait to
+	// return with `tryAnother==true` then a following call to Load
+	// will return X or an even more recently stored value.
+	for _, plState := range newlyQuiescent {
+		plState.queues.Quiesce(plState.emptyHandler)
+	}
+}
+
+func qscOfPL(pl *rmtypesv1a1.PriorityLevelConfiguration, requestWaitLimit time.Duration) (fq.QueueSetConfig, error) {
+	qsConfig := fq.QueueSetConfig{Name: pl.Name,
+		RequestWaitLimit: requestWaitLimit}
+	var err error
+	if qc := pl.Spec.Limited.LimitResponse.Queuing; qc != nil {
+		dealer, e1 := shufflesharding.NewDealer(int(qc.Queues), int(qc.HandSize))
+		if e1 != nil {
+			err = errors.Wrap(e1, fmt.Sprintf("priority level %q has QueuingConfiguration %#+v, which caused shufflesharding.NewDealer to fail", pl.Name, *qc))
+		}
+		qsConfig.DesiredNumQueues = int(qc.Queues)
+		qsConfig.QueueLengthLimit = int(qc.QueueLengthLimit)
+		qsConfig.Dealer = dealer
+	}
+	return qsConfig, err
+}
+
+func (reqMgr *requestManager) syncFlowSchemaStatus(fs *rmtypesv1a1.FlowSchema, isDangling bool) {
+	danglingCondition := apihelpers.GetFlowSchemaConditionByType(fs, rmtypesv1a1.FlowSchemaConditionDangling)
+	if danglingCondition == nil {
+		danglingCondition = &rmtypesv1a1.FlowSchemaCondition{
+			Type: rmtypesv1a1.FlowSchemaConditionDangling,
+		}
+	}
+
+	switch {
+	case isDangling && danglingCondition.Status != rmtypesv1a1.ConditionTrue:
+		danglingCondition.Status = rmtypesv1a1.ConditionTrue
+		danglingCondition.LastTransitionTime = metav1.Now()
+	case !isDangling && danglingCondition.Status != rmtypesv1a1.ConditionFalse:
+		danglingCondition.Status = rmtypesv1a1.ConditionFalse
+		danglingCondition.LastTransitionTime = metav1.Now()
+	default:
+		// the dangling status is already in sync, skip updating
+		return
+	}
+
+	apihelpers.SetFlowSchemaCondition(fs, *danglingCondition)
+
+	_, err := reqMgr.flowcontrolClient.FlowSchemas().UpdateStatus(fs)
```

Comment:
we'd better wrap it w/ `retry.OnConflict` so that it retries on resourceVersion conflict, and the function should return an error on failure. (consider this in a follow-up)

Reply:
Hmm, yes, there is an issue there.
On further thought, there is also another issue regarding these update calls: in the call to `digestConfigObjects` made from the function that constructs a `requestManager`, the server is not even ready to handle requests yet.